### PR TITLE
Refactor execution errors

### LINKED ISSUE #197: Refactor execution errors
There are a lot of execution errors in the AquaVM and some of them have a "real" runtime nature, but some could be found at the validation stage. Since we are planning to use execution error codes in match/mismatch to support monadic error flow in Aqua, it needs to getting rid of all errors that could be checked at the parsing stage and using their error code are unsound.

### PR DESCRIPTION
**Pull request by mikevoronov**
Resolves #197.

### CODE REVIEW

**Review by ValeryAntopol — APPROVED** [review]
The issues found are cosmetic except one big question arising from my not imperfect understanding of AquaVM, so i approve the merge of the changes after cosmetics fixed and the question solved (which is most likely "everething is okay" or the solution is obvious). Also we need to bump minor version, as the public api — error codes — is changed.

File: air/src/execution_step/errors/catchable_errors.rs
Comment (mikevoronov) [reply air/src/execution_step/errors/catchable_errors.rs:43]: As I already tols you `VariableNotFound` is used to support the fork-join model (join behaviour in our terms) and it's absolutely OK if some variable hasn't been defined yet, due to the par nature of AIR. But if it confuses you so much, I'll add a comment.

File: air/tests/test_module/integration/flattening.rs
Comment (mikevoronov) [reply air/tests/test_module/integration/flattening.rs:79]: Actually it's checked internally by `checked_vm_call`, but I automatically remove it and forget why it's explicit - just to avoid such confuses.

File: air/src/execution_step/boxed_value/stream.rs
Comment (mikevoronov) [reply air/src/execution_step/boxed_value/stream.rs:34]: It's because `ExecutionError` became public and `Stream` is used inside one of it's variants. And `ExecutionError` became public to use it in tests.

Btw, after boxed value landed, `Stream`'ll in a separate crate and must be public anyway, so it's all ok.

File: air/src/execution_step/errors/catchable_errors.rs
Comment (mikevoronov) [reply air/src/execution_step/errors/catchable_errors.rs:94]: Just in case, since it's a private mapping no one will hurt by that.

File: crates/air-lib/air-parser/src/ast/values/impls.rs
Comment (mikevoronov) [reply crates/air-lib/air-parser/src/ast/values/impls.rs:91]: It's unnecessary, because it's used by methods that didn't required this lifietime declared explicitly, but after validator refactoring it became necessary.

File: air/src/execution_step/errors/uncatchable_errors.rs
Comment (mikevoronov) [reply air/src/execution_step/errors/uncatchable_errors.rs:62]: Yep, it's a good idea.

**Comment by mikevoronov** [comment]
Error codes aren't a public API at the moment, they just used inside tests and for debugging purposes.

**Pull request merged**
